Snowflake.Data/Core/SFStatement.cs

```diff
@@ -356,12 +356,18 @@ internal SFBaseResultSet Execute(int timeout, string sql, Dictionary<string, Bin
                              bindings,
                              describeOnly);
 
+                    if (logger.IsDebugEnabled())

```

Comment:
is it a kind of optimisation? Debug operation has this check in itself, so I would rather not duplicate it here. Especially that then the debug log could not be covered by the tests so it may decrease the code coverage.

Reply:
I was thinking about not loosing a sec if debug is not enabled but the string being built is not that huge, I'll change it